transformers/tokenization_utils.py

```diff
@@ -718,6 +729,12 @@ def encode(self,
                 - 'only_first': Only truncate the first sequence
                 - 'only_second': Only truncate the second sequence
                 - 'do_not_truncate': Does not truncate (raise an error if the input sequence is longer than max_length)
+            padding_strategy: if set to a strategy, the returned sequences will be padded according to the model's 
+                padding index, up to their max length. If no max length is specified, no padding is done.
```

Comment:
In the most general case for the user, we may want to allow passing a batch of sequence and just have them padded to the maximal sequence length in the batch (or probably the min between the model's max length and the maximal sequence length in the batch).
Should we prepare for that?

Reply:
I feel like we should do padding up to `self.max_length` if padding is asked even when no explicit `max_length` is provided to the method.

Reply:
Regarding the first comment, we don't currently manage sequence batching so it would require implementing an `encode_batch` method or similar beforehand, wouldn't it? As of right now, it shouldn't be too hard for the user to identify the longer sequence in its batches, but it would be a good idea to ease the process of tokenizing batches.

Totally agree with your second comment, this makes sense!